extension/src/plots/model/collect.ts

```diff
@@ -282,69 +295,20 @@ const collectCustomPlotData = (
 }
 
 export const collectCustomPlotsData = (
-  metricsAndParams: CustomPlotsOrderValue[],
+  plotsOrderValues: CustomPlotsOrderValue[],
+  checkpointPlots: CustomCheckpointPlots,
   experiments: Experiment[]
-): CustomPlotData[] => {
-  return metricsAndParams.map(({ metric, param }) =>
-    collectCustomPlotData(metric, param, experiments)
-  )
-}
-
-type MetricOrderAccumulator = {
-  newOrder: string[]
-  uncollectedMetrics: string[]
-  remainingSelectedMetrics: string[]
-}
-
-const collectExistingOrder = (
-  acc: MetricOrderAccumulator,
-  existingMetricOrder: string[]
-) => {
-  for (const metric of existingMetricOrder) {
-    const uncollectedIndex = acc.uncollectedMetrics.indexOf(metric)
-    const remainingIndex = acc.remainingSelectedMetrics.indexOf(metric)
-    if (uncollectedIndex === -1 || remainingIndex === -1) {
-      continue
-    }
-    acc.uncollectedMetrics.splice(uncollectedIndex, 1)
-    acc.remainingSelectedMetrics.splice(remainingIndex, 1)
-    acc.newOrder.push(metric)
-  }
-}
-
-const collectRemainingSelected = (acc: MetricOrderAccumulator) => {
-  const [newOrder, uncollectedMetrics] = splitMatchedOrdered(
-    acc.uncollectedMetrics,
-    acc.remainingSelectedMetrics
-  )
-
-  acc.newOrder.push(...newOrder)
-  acc.uncollectedMetrics = uncollectedMetrics
-}
-
-export const collectMetricOrder = (
-  checkpointPlotData: CheckpointPlot[] | undefined,
-  existingMetricOrder: string[],
-  selectedMetrics: string[] = []
-): string[] => {
-  if (!definedAndNonEmpty(checkpointPlotData)) {
-    return []
-  }
-
-  const acc: MetricOrderAccumulator = {
-    newOrder: [],
-    remainingSelectedMetrics: [...selectedMetrics],
-    uncollectedMetrics: checkpointPlotData.map(({ id }) => id)
-  }
-
-  if (!definedAndNonEmpty(acc.remainingSelectedMetrics)) {
-    return acc.uncollectedMetrics
-  }
-
-  collectExistingOrder(acc, existingMetricOrder)
-  collectRemainingSelected(acc)
-
-  return [...acc.newOrder, ...acc.uncollectedMetrics]
+): CustomPlot[] => {
+  return plotsOrderValues
+    .map((plotOrderValue): CustomPlot => {
+      if (isCheckpointValue(plotOrderValue)) {
```

Comment:
Since CustomPlot has two types, we end up needing to this kind of check multiple times throughout  code, but is there a better way to handle the different types?

Reply:
Maybe adding a boolean value to `plotOrderValue` to easily check. Do we actually have to keep them separated throughout the code if the only difference is that one has a `param` property and not the other?